src/shared/components/query/GenesetsHierarchySelector.tsx

```diff
@@ -0,0 +1,112 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import * as styles_any from './styles/styles.module.scss';
+import {action, ObservableMap, expr, toJS, computed, observable} from "mobx";
+import {observer, Observer} from "mobx-react";
+import {MutSig} from "../../api/generated/CBioPortalAPIInternal";
+import ReactSelect from 'react-select';
+import GenesetsJsTree from "./GenesetsJsTree";
+import { InputHTMLAttributes, FormEvent } from "react";
+
+const styles = styles_any as {
+    GenesetsSelectorWindow: string,
+    selectionColumnHeader: string,
+    selectionColumnCell: string,
+};
+
+export interface GenesetsHierarchySelectorProps
+{
+    initialSelection: string[];
+    gsvaProfile: string;
+    sampleListId: string|undefined;
+    onSelect: (map_geneSet_selected:ObservableMap<boolean>) => void;
+}
+
+@observer
+export default class GenesetsHierarchySelector extends React.Component<GenesetsHierarchySelectorProps, {}>
+{
+    @observable appliedPercentile: string;
+    @observable appliedPvalueThreshold: string;
+    @observable appliedScoreThreshold: string;
+    @observable percentileSelectOptions: {label: string, value: string};
+    @observable pvalueThreshold: string;
+    @observable scoreThreshold: string;
+    @observable searchValue: string;
+    @observable triggerGeneSetSelection: boolean;
+    readonly percentileOptions = [{label: '50%', value: '50'}, {label: '75%', value: '75'}, {label: '100%', value: '100'}];
+    constructor(props:GenesetsHierarchySelectorProps)
+    {
+        super(props);
+        this.appliedPercentile = '75';
+        this.appliedPvalueThreshold = "0.05";
+        this.appliedScoreThreshold = "0.5";
+        this.percentileSelectOptions = {label: '75%', value: '75'};
+        this.pvalueThreshold = "0.05";
+        this.scoreThreshold = "0.5";
+        this.searchValue = "";
+        this.percentileChange = this.percentileChange.bind(this);
+        this.applyFilter = this.applyFilter.bind(this);
+    }
+    
+    percentileChange(val: {label: string, value: string} | null)
+    {
+        this.percentileSelectOptions = val || {label: '75%', value: '75'};
+    }
+    
+    applyFilter() {
+        this.appliedPercentile = this.percentileSelectOptions.value;
+        this.appliedPvalueThreshold = this.pvalueThreshold;
+        this.appliedScoreThreshold = this.scoreThreshold;
+    }
+    
+    render()
+    {
+        return (
+                <div>
```

Comment:
see if you can get better indentation on the markup.  IDE code formatting should work.  I use WebStorm and will often run the formatter

Reply:
And to make the structure of this long block of JSX even clearer, might some section of it make sense as a component on its own? [Here](https://reactjs.org/docs/components-and-props.html#extracting-components)'s a guide on how to extract components out of complex components, in case it does.

Reply:
As suggested by @fedde-s, I isolated the component `GenesetsHierarchyFilterForm`. I think now the whole JSX code is more readable.